core/deployment/src/main/java/io/quarkus/runner/bootstrap/StartupActionImpl.java

```diff
@@ -66,12 +72,47 @@ public StartupActionImpl(CuratedApplication curatedApplication, BuildResult buil
             runtimeClassLoader = baseClassLoader;
         }
         this.runtimeClassLoader = runtimeClassLoader;
+        handleEagerClasses(runtimeClassLoader, eagerClasses);
+    }
+
+    private void handleEagerClasses(QuarkusClassLoader runtimeClassLoader, Set<String> eagerClasses) {
+        int availableProcessors = Runtime.getRuntime().availableProcessors();
+        if (availableProcessors == 1) {
+            return;
+        }
+        //leave one processor for the main startup thread
+        ExecutorService loadingExecutor = Executors.newFixedThreadPool(availableProcessors - 1);
```

Comment:
What if you have only one core in a container?

Reply:
This code is not run, there is a check 4 lines above that will bail out.

Reply:
Hum. Silly me.